src/compiler/emitter.ts

```diff
@@ -55,6 +55,20 @@ module ts {
     }
 
     let indentStrings: string[] = ["", "    "];
+
+    // When emitting for ES3, property access via dot notation (i.e. obj.prop ) cannot be used
+    // if the property name is a reserved word.  Use index notation (i.e. obj["prop"]).
+    let ES3ReservedWords = [

```

Comment:
After cross-checking the list, the following are all ES3 FutureReservedWords, but not currently in the SyntaxKind list, which means we'd need to add them to SyntaxKind (and the textToTokenMap), even though we'll never parse/create a node of these kind. That seems less clean than the current design. Thoughts?

ES3 reserved words we currently don't have SyntaxKind members for: "abstract, int, short, boolean, byte, long, char, final, native, synchronized, float, throws, goto, transient, volatile"

Note: These are not keywords or reserved in ES5+